### PR TITLE
Restructure / clean up project a lot

### PR DESCRIPTION
**Pull request by arthurtilly**
* New games/ folder containing various data related to each game engine

* Game engine, prologue engines/scripts, game/prologue graphics, game text moved into new folders

* Lots of .rodata converted into .data segments and const qualifiers removed

* Renaming of some stuff to be more consistent

* Graphics header files added to remove excess externs

* Makefile cleaned up

### CODE REVIEW

File: data/beatscript_scene_data.c [hunk 1]
```diff
@@ -2,17 +2,17 @@
 #include "beatscript_scene_data.h"
 
 
-/* BeatScript Scene Data */
+/* Beatscript Scene Data */
```
Comment (patataofcourse) [inline data/beatscript_scene_data.c:5]: Any reason why we're going with "Beatscript" instead of "BeatScript"?

**Review by patataofcourse — COMMENTED** [review]
Seems fine to me. Bit iffy on Ninja Reincarnate tho, since it's one of the more divisive names Tengoku-wise.

File: data/beatscript_scene_data.c (hunk 1 above)
Comment (arthurtilly) [reply data/beatscript_scene_data.c:5]: I mean it's not called TickFlow, I'm not sure why it was ever capitalised like that to begin with

**Comment by arthurtilly** [comment]
> Seems fine to me. Bit iffy on Ninja Reincarnate tho, since it's one of the more divisive names Tengoku-wise.

Did think about this but I searched it up in the server and Reincarnate was much more commonly used than Ninja Bodyguard 2 (neither were incredibly common tbf), and I could not find a single piece of discourse about the name.

**Comment by arthurtilly** [comment]
and even in the RH Advance server there does seem to be a lot of people who like the name Reincarnate tbf, and personally I agree

File: data/beatscript_scene_data.c (hunk 1 above)
Comment (patataofcourse) [reply data/beatscript_scene_data.c:5]: fair enough

**Comment by patataofcourse** [comment]
kk, just wanted to mention that

**Comment by GenericArrangements** [comment]
I'm not really a fan of "Reincarnate" as a substitute for a word that means "Descendant" (because they're not the same thing!), but that's not really significant enough to hold back this pull request or anything. I've also grown used to the ``games`` folder structure now, though it is rather odd seeing Mr. Upbeat lacking a graphics folder (I know why it does - it's just a bit odd to look at). A suggestion like "put Mr. Upbeat in the same folder as Metronome" seems very silly and even less significant though.

Anyway, uh... yeah this is good I think. Nice to do away with most of those **``const``** labels.

**Pull request merged**